internal/app/machined/pkg/controllers/k8s/control_plane_static_pod.go

```diff
@@ -429,6 +429,29 @@ func (ctrl *ControlPlaneStaticPodController) manageAPIServer(ctx context.Context
 		"service-account-signing-key-file": argsbuilder.MergeDenied,
 		"tls-cert-file":                    argsbuilder.MergeDenied,
 		"tls-private-key-file":             argsbuilder.MergeDenied,
+		"authorization-config":             argsbuilder.MergeDenied,
+	}
+
+	k8sVersion := compatibility.VersionFromImageRef(cfg.Image)
+
+	if k8sVersion.KubeAPIServerSupportsAuthorizationConfigFile() {
+		if _, ok := cfg.ExtraArgs["authorization-mode"]; ok {
+			return "", fmt.Errorf("authorization-mode is not supported when using authorization-config")
+		}
+
+		for flag := range cfg.ExtraArgs {
+			if strings.HasPrefix(flag, "authorization-webhook-") {
+				return "", fmt.Errorf("%s flag is not supported when using authorization-config", flag)
+			}
+		}
+
+		if !k8sVersion.FeatureFlagStructuredAuthorizationConfigurationEnabledByDefault() {
+			builder.Set("feature-gates", "StructuredAuthorizationConfiguration=true")
```

Comment:
I guess we need to enable appending to the feature-gates? or is it fine?

Reply:
it's already defined as `argsbuilder.MergeAdditive`, updated tests to reflect the same